### PR TITLE
add pylint to precommit

### PR DESCRIPTION
**Pull request by JudahSchwartz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue
](https://jira-dc.paloaltonetworks.com/browse/CIAC-7971)
## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -35,9 +35,13 @@ jobs:
...
 
     - name: run pre-commit
+      env:
+        DOCKERHUB_USER: ${{ secrets.DOCKERHUB_USER }}
+        DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
       run: |
         source .venv/bin/activate
-        demisto-sdk pre-commit -g --unit-test --validate --no-secrets --show-diff-on-failure --verbose --mode=nightly
+        docker pull demisto/boto3py3:1.0.0.79776
+        demisto-sdk pre-commit -a --unit-test --validate --no-secrets --show-diff-on-failure --verbose --mode=nightly
```
Comment (JudahSchwartz) [inline .github/workflows/pre-commit.yml:44]:
```suggestion
        demisto-sdk pre-commit -g --validate --no-secrets --show-diff-on-failure --verbose --mode=nightly
```

**Comment by JudahSchwartz** [comment]
```
Error: The file .pylintrc cannot be deleted. Please restore the file.
Error: The file Tests/scripts/pylint.sh cannot be deleted. Please restore the file.
```
will need a force

File: .pre-commit-config_template.yaml [hunk 2]
```diff
@@ -43,6 +43,20 @@ repos:
   - id: autopep8
 - repo: local
   hooks:
+
+  - id: pylint-in-docker
+    name: pylint-in-docker
+    description: my description
```
Comment (JudahSchwartz) [inline .pre-commit-config_template.yaml:49]:
```suggestion
    description: Run pylint on the code in content packs
```

File: .pre-commit-config_template.yaml [hunk 3]
```diff
@@ -43,6 +43,20 @@ repos:
...
+    description: Run pylint on the code in content packs
+    docker_image:nightly: from-yml
+    docker_image: native:ga,from-yml,native:dev,native:maintenance
+    entry: pylint
+    files: Packs\/.*\.py$
+    exclude: (_test\.py)|(.vulture_whitelist\.py)$
+    args: ['--ignore=demistomock.py,CommonServerPython.py,CommonServerUserPython.py,conftest.py,.venv', '-E', '--disable=bad-option-value,unsubscriptable-object', '-d duplicate-string-formatting-argument', "--msg-template='{path}:{line}:{column}: {msg_id} {obj}: {msg}'", '--generated-members=requests.packages.urllib3,requests.codes.ok']
+    config_file_arg:
+      arg_name: '--rcfile'
+      file_name: '.pylintrc'
```
Comment (ShahafBenYakir) [inline .pre-commit-config_template.yaml:58]: Why do we need to have this data if we are deleting the file?

**Review by ShahafBenYakir — COMMENTED** [review]
some questions

File: .pre-commit-config_template.yaml (hunk 3 above)
Comment (JudahSchwartz) [reply .pre-commit-config_template.yaml:58]: this is for the ones in the pack directories

File: Tests/scripts/pylint.sh [hunk 4]
```diff
@@ -1,22 +0,0 @@
-#!/bin/bash
-
-pylint_disabled_errors=C0103,C0114,C0115,C0116,C0122,C0301,C0302,C0325,C0411,C0412,C0413,C0415,E1136,E1205,F0001,F0010,R0201,R0205,R0401,R0801,R0902,R0903,R0904,R0912,R0913,R0914,R0915,R1702,R1705,R1710,R1721,R1725,W0105,W0150,W0212,W0401,W0404,W0511,W0603,W0612,W0613,W0621,W0622,W0703,W1202,W1203
```
[the remark this answers is not in the document]
Comment (JudahSchwartz) [reply Tests/scripts/pylint.sh:3]: these rules were only for pylint run that i just removed

**Pull request merged**